libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowContainer.m

```diff
@@ -0,0 +1,289 @@
+/*
+ SFSDKWindowContainer.m
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on 7/4/17.
+ 
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#import "SFSDKWindowContainer.h"
+
+@interface SFSDKWindowContainer() {
+    UIViewController *_modalViewController;
+}
+@property (nonatomic, strong) NSMutableOrderedSet *delegates;
+@end
+
+@implementation SFSDKWindowContainer
+@synthesize window = _window;
+
+- (instancetype)initWithWindow:(UIWindow *)window andName:(NSString *) windowName {
+    
+    self = [super init];
+    if (self) {
+        _delegates = [NSMutableOrderedSet orderedSet];
+        _window = window;
+        _windowName = windowName;
+    }
+    return self;
+}
+
+- (UIWindowLevel)windowLevel{
+    return self.window.windowLevel;
+}
+
+- (void)setWindowLevel:(UIWindowLevel)windowLevel {
+    if (self.window.windowLevel != windowLevel){
+        self.window.windowLevel = windowLevel;
+    }
+}
+
+- (void)pushViewController:(UIViewController *)controller {
+    [self pushViewController:controller animated:NO completion:nil];
+}
+
+- (void)popViewController:(UIViewController *)controller {
+    [self popViewController:controller animated:NO completion:nil];
+}
+
+- (void)pushViewController:(UIViewController *)viewController animated:(BOOL)flag completion:(void (^)(void))completion {
```

Comment:
I'm not a big fan of these methods.  The idea of establishing a multi-window hierarchy was in part to get rid of fancy view controller management, in favor of a more established best practices for consumers to manage their own view hierarchies.

In this paradigm, I feel that we should leave view controller management to consumers of this functionality.  Even for internal consumption, we should know enough about our view hierarchies to explicitly manage what gets shown, where.  And if we don't, we need to ask why.

Leaving this functionality around encourages the kind of deviation from best practices that we've historically seen in this realm.  Consumers should be responsible for their view state management.